### DISCUSSION TITLE
Tutorial on block deprecation

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#67] [Topic Ideas]
If you make a change to the way that a block is rendered you will see the "invalid content" error message after the change.

![this-block-contains-unexpected-or-invalid-content](https://user-images.githubusercontent.com/20643925/214892339-0cf4d378-c1db-4ca9-bbe4-c5eb7ff6ea01.jpg)

This commonly occurs during development, and developers working with blocks are very aware of this and are used to dealing with it.

However, if you've already published a block and want to make changes to it in an update you won't want users to see the "invalid content" error message, which they will in each instance of the block across their site until recovery is done on each of them.

To circumvent this problem developers can "deprecate" the old version of the block.

A tutorial on how to deprecate a block, and when and why you should do it, would be a nice addition to the Developer Blog.

Reference:
[Deprecation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/)

**Comment by mburridge** [comment]
🙋‍♂️ Flagging that I'd be happy to take on authorship of this, so I haven't added the "needs writer" label.

**Comment by bph** [comment]
That's a great topic @mburridge - I feel many block developers shy away entirely  just because it's not well understood and create dynamic blocks instead.

**Comment by bph** [comment]
This is now in the works. #71 and this discussion is locked. Please continue discussion on the issue.

**Comment by bph** [comment]
Published: [Block deprecation – a tutorial](https://developer.wordpress.org/news/2023/03/block-deprecation-a-tutorial/)
